code/game/machinery/machinery.dm

```diff
@@ -348,6 +348,22 @@
 		reregister_machine()
 		power_change()
 
+/obj/machinery/attackby(obj/item/M, mob/user, params) // Scanning the machine with a machine analyser
+	if(istype(M, /obj/item/robotanalyzer))
+		user.visible_message("<span class='notice'>[user] scans [src] with [M].</span>",\
+		"<span class='notice'>You scan [src] with [M]</span>")
+		if(obj_integrity == max_integrity)
+			to_chat(user, "<span class='info'>[src] is at full integraty.</span>")
+			return
+		if(stat & BROKEN)
+			to_chat(user, "<span class='warning'>Catastrophic structural damage detected! [src] requires complete reconstruction.</span>")
+			return
+		else
+			to_chat(user, "<span class='info'>Structural damage detected! [src]'s estimated integraty is [round((obj_integrity/max_integrity)*100)]%.</span>")
+			return
+	else
+		return ..()
+
```

Comment:
This should probably be in the machine analyzer's `attack()` instead
Currently, that proc only assumes input of a living mob and runs `robot_healthscan()` on it, but we could instead make it accept any atom and run `robot_healthscan()`/a new `machine_scan()` or sth, according to what was smacked

Reply:
Yeah, the brunt of this behavior should probably be a function of the analyzer more than the machine